pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/QueryRunnerTest.java

```diff
@@ -109,6 +109,23 @@ private Object[][] provideTestSqlAndRowCount() {
         new Object[]{"SELECT a.col1, a.ts, b.col2, b.col3 FROM a JOIN b ON a.col1 = b.col2 "
             + " WHERE a.col3 >= 0 AND a.col2 = 'alice' AND b.col3 >= 0", 3},
 
+        // Join query with IN and Not-IN clause. Table A's side of join will return 9 rows and Table B's side will
+        // return 2 rows. Join will be only on col1=bar and since A will return 3 rows with that value and B will return
+        // 1 row, the final output will have 3 rows.
+        new Object[]{"SELECT a.col1, b.col2 FROM a JOIN b ON a.col1 = b.col1 "
+            + " WHERE a.col1 IN ('foo', 'bar', 'alice') AND b.col2 NOT IN ('foo', 'alice')", 3},
```

Comment:
can you also add a test in QueryCompilationTest (QueryEnvironmentTestBase) for the parsing? just one single entry should be enough by mixing all types of predicates together

Reply:
Done. Also added test for empty IN clause and range predicates to verify they are failing. The range test I'll move to testPlanWithoutException in the range PR.